hpx/runtime/components/client_base.hpp

```diff
@@ -289,11 +289,18 @@ namespace hpx { namespace components
         }
 
         ///////////////////////////////////////////////////////////////////////
+#if defined(HPX_HAVE_COMPONENT_GET_GID_COMPATIBILITY)
         naming::id_type const & get_gid() const
         {
             return gid_.get();
         }
 
+#endif
+        naming::id_type const & get_id() const
+        {
+            return gid_.get();
+        }
+
```

Comment:
On one hand, you are proposing to get rid of id_types altogether, and now you're suggesting to expose unmanaged id_types to the user (for questionable optimizations). That is a contradiction which does not make any sense to me, sorry.